### ISSUE TITLE
CodeNarcServer should listen on localhost only

### ISSUE THREAD

**Issue by maurofaccenda** [nvuillam/npm-groovy-lint#56]
Hi, I believe for security reasons, by default the server should listen on localhost only. Listening on any interface could unnecessarily expose the users' system.

**Comment by nvuillam** [comment]
This is totally true, and whereas the risk to have someone force the linting of groovy files from outside is low, this is a problem that should be solved.
npm-groovy-lint v1 was using local socket connection, but it didn't work on mac and linux, that's why i used "open" port so that every computer was compliant.
I see you forked the repo, are you working on it, or should I add that to my todo list ? 😅

**Comment by nvuillam** [comment]
This is the solution i initially implemented but was not mac/linux compliant

https://stackoverflow.com/questions/50770747/how-to-configure-com-sun-net-httpserver-to-accept-only-requests-from-localhost

**Comment by maurofaccenda** [comment]
On my scenario, the computer have policies that doesn't allow any process to listen on any 'external' network interface. So the CodeNarcServer fails to run. To overcome that locally, I've changed the options.js to use `http://127.0.0.1` as default `serverhost`.

That probably would work on any system (I tested on Mac), except if it has IPv4 disabled (if that's the case it should use `http://[:]`.

If you want to find a better approach, I am willing to try on both Mac and Linux and make sure it works. :)

**Comment by maurofaccenda** [comment]
> I see you forked the repo, are you working on it, or should I add that to my todo list ? 😅

I'm no JS dev. But I if the suggestion I've made just before works for you, I can create a PR.

Btw, I forgot to thank you for the projects, I've been looking for something like that for quite some time :)

**Comment by nvuillam** [comment]
I'll try to make such update in a beta version and ask you to try :)

I have CI on Windows & Linux, but for mac CircleCI is paying so your help will be welcome :)

**Comment by nvuillam** [comment]
I tried [an update](https://github.com/nvuillam/npm-groovy-lint/pull/59) and it seems to be [ok on Windows and Linux](https://app.circleci.com/jobs/github/nvuillam/npm-groovy-lint/630)

Please can you try the following, and confirm that :  
- it works :)
- CodeNarcServer is called and performances are not impacted compared to current released version

```shell
npm install -g npm-groovy-lint@5.2.0-beta.0
npm-groovy-lint --killserver
npm-groovy-lint
```

Many thanks :)

**Comment by nvuillam** [comment]
@maurofaccenda , any news ? :)

**Comment by maurofaccenda** [comment]
@nvuillam hey. i've really busy this week. i'll give it a try as soon as i can get some spare time to try it (my work env -- where i have a mac -- is really strict, so it takes some effort to run stuff like this). i'll give a shout as soon as possible.

**Comment by nvuillam** [comment]
Thanks :)

Le mar. 30 juin 2020 à 13:07, Mauro Faccenda <notifications@github.com> a
écrit :

**Comment by maurofaccenda** [comment]
I've managed to test. It worked out of the box. But it seem to listen on all interfaces (0.0.0.0:7484).

**Comment by nvuillam** [comment]
Thanks for your test ! It works on Windows & Linux too, so ... let's have a leap of faith :)

I implemented https://stackoverflow.com/questions/50770747/how-to-configure-com-sun-net-httpserver-to-accept-only-requests-from-localhost , 4 likes for the answer, and the author seems to be a java expert and has a very high reputation on stackexchange (best 0.01%) so let's trust him :D 

It seems that listen != accept, so even if it listens to maybe more than localhost, it will accept only from localhost

But if you hear of a stronger way, don't hesitate to reopen the issue :)

**Comment by nvuillam** [comment]
Published in v5.4.0 :)
